### PR TITLE
Fix moe cpu offload

### PR DESCRIPTION
**Pull request by RezaYazdaniAminabadi**
The MoE- param gradients norms don't need to be averaged when created on CPU only when using 1-DP training. However, I just moved the tensor back to GPU to get average when having data-parallel on the MoE parameters and using CPU-offload.

This PR addresses https://github.com/microsoft/DeepSpeed/issues/5203

### CODE REVIEW

File: deepspeed/runtime/zero/stage_1_and_2.py [hunk 1]
```diff
@@ -1782,7 +1782,7 @@ def scaled_global_norm(self, norm_type=2):
             else:
                 norm_groups.append(self.get_grad_norm_direct(self.averaged_gradients[i], self.params_in_partition[i]))
 
-        if self.has_moe_layers:
+        if self.has_moe_layers and self.device != 'cpu':
```
Comment (tjruwase) [inline deepspeed/runtime/zero/stage_1_and_2.py:1785]: @RezaYazdaniAminabadi, won't this result in different gradient norms printout for offload vs non-offload?

Comment (RezaYazdaniAminabadi) [reply deepspeed/runtime/zero/stage_1_and_2.py:1785]: as long as the same global batch size is used between the two, I think they should stay the same. When non-offload, if one GPU is used then there is no averaging and when multi-gpu used then the batch is partitioned (assuming we have data-parallel for the expert part), and then only a partition of the data is used per rank and potentially the norm should the same result when reduced. Is my understanding correct?

Comment (tjruwase) [reply deepspeed/runtime/zero/stage_1_and_2.py:1785]: I think the right solution would be to create a per-rank device tensor (e.g., self.scaled_gradient_norm) to represent this [value](https://github.com/RezaYazdaniAminabadi/DeepSpeed/blob/9a59df40e7b693acb968f2d14c8b65cc913eeea7/deepspeed/runtime/zero/stage_1_and_2.py#L1948) and update during accumulation, and then allreduce before step.

Comment (RezaYazdaniAminabadi) [reply deepspeed/runtime/zero/stage_1_and_2.py:1785]: I see, you are worried that this might cause issue in the case of multi-gpu that still we have the CPU-offload and the data between ranks are different?

Comment (RezaYazdaniAminabadi) [reply deepspeed/runtime/zero/stage_1_and_2.py:1785]: okay, will make the change and commit shortly

Comment (RezaYazdaniAminabadi) [reply deepspeed/runtime/zero/stage_1_and_2.py:1785]: thanks @tjruwase

Comment (tjruwase) [reply deepspeed/runtime/zero/stage_1_and_2.py:1785]: Thanks @RezaYazdaniAminabadi

**Review by tjruwase — APPROVED** [review]
